Initial content (not yet complete) (#6)
git-bom/site: content/_index.md

---
title: Home
---
{{< home/section title="What?" iconclass="fas fa-2x fa-power-off">}}
GitBOM is neither [git](/glossary/#git) nor an [SBOM](/resources/glossay/#sbom).
<!-- 
GitBOM is an ultra simple scheme for [primary build tools](/resources/glossay/#primary-build-tools) to embed into the artifact being built a
short simple verifiable reference to complete artifact tree of that artifact with absolute precision all the way down to the source code. -->

> GitBOM is an ultra simple scheme for a [build tool](/glossary/build_tool) to:

Reviewer: ```suggestion
GitBOM is a minimalistic scheme for [build tools](/glossary/build_tool) to:
```

> 1. Construct a complete [artifact tree](/glossary/artifact_tree) with absolute precision all the way down to the individual source code files incorporated into the [artifact](/glossary/artifact) being built.

Reviewer: ```suggestion
1. Build a compact [artifact tree](/glossary/artifact_tree), tracking every source code file incorporated into each built [artifact](/glossary/artifact).
```

2. Embed an id for that artifact tree into the artifact being built.

> in a language-heterogenous environment with zero developer effort, involvement, or awareness.

Reviewer: ```suggestion
3. in a language-heterogenous environment, across packaging formats, with zero developer effort, involvement, or awareness.
```

> - GitBOM is not an SBOM, though it is designed to complement existing SBOM efforts such as [SPDX](https://spdx.dev/), [CycloneDX](https://cyclonedx.org/), or [SWID](https://nvd.nist.gov/products/swid).

Reviewer: ```suggestion
- GitBOM is not an SBOM. It is designed to complement SBOMs, such as [SPDX](https://spdx.dev/), [CycloneDX](https://cyclonedx.org/), or [SWID](https://nvd.nist.gov/products/swid).
```

> - GitBOM is not a version control system. It is designed to co-exist with them, but not require them.

Reviewer: ```suggestion
- GitBOM is not a version control system. It is designed to co-exist with, but not require, them.
```

> GitBOM applies the [Unix Philosophy](https://en.wikipedia.org/wiki/Unix_philosophy). It does one thing well: constructing [artifact trees](/glossary/artifact_tree).

Reviewer: ```suggestion
GitBOM applies the [Unix Philosophy](https://en.wikipedia.org/wiki/Unix_philosophy) of "do one thing, and do it well." GitBOM constructs [artifact trees](/glossary/artifact_tree).
```

<!-- GitBOM borrows the use of [merkle tree](/resources/glossay/#merkle-tree) and the [gitref](/resources/glossay/#merkle-tree) from [git](/resources/glossay/#mgit) to construct a [verifiable bare artifact tree](/resources/glossay/#verifiable-bare-artifact-tree) in a manner suitable for [primary build tools](/resources/glossay/#primary-build-tools) to compute the GitBOM for an artifact and embed a unqiue, immutable, verifiable GitBOM id in the artifact being built. 
in language-heterogenuous environments with zero-end-user-effort. -->

{{< /home/section >}}
<!-- 
{{< home/section title="GitBOM is not:" iconclass="fas fa-2x fa-power-off">}}
1. Not a system for build reproducability, but it does provide information that is useful for that.
2. Not a version control system, though it is designed to co-exist with them.
3. Not an SBOM, though it is designed to complement them.
4. Not a globally unique software identifier (SWID).
5. Not reliant on any particular packaging or distribution mechanism, either for artifacts or for artifact identity graphs).
{{< /home/section >}} -->

> {{< home/section title="Why?" >}}

Reviewer: I've left a few specific suggestions below, but my gut says this still isn't the right impact. WDYT about the "WHY" section being a one-paragraph explanation of how GitBOM _would have helped_ identify log4j?

Something like this...

> By correlating every piece of software to a verifiable, complete, and concise artifact tree, GitBOM would have enabled enterprises to easily and independently identify which networked components (both open source and proprietary) contained the `log4j` library. GitBOM would allow an organization to reliably cross-reference all software they consume against CVE databases.

Author: Ah... interesting... more narrative :)  Will see what I can do there :)

> A complete artifact tree with absolute precision all the way down to the individual source code files enables:

Reviewer: ```suggestion
By correlating every piece of software to a verifiable, complete, and concise artifact tree, we enable: 
```

> - Detecting potential vulnerabilities in executables/containers.

Reviewer: ```suggestion
- Run-time detection of potential vulnerabilities in executables/containers by scanning a "fingerprint" rather than the contents.
```

> - Identifying Open Source License obligations

Reviewer: ```suggestion
```

> - Identifying commercial license obligations

Reviewer: ```suggestion
- Identifying license obligations across both open source and commercial software
```

- More reliable attestation

> - Post exploit forensics

Reviewer: ```suggestion
- Improved accuracy when cross-referencing public vulnerability databases
```

Author: Do we really want to eliminate 'post exploit forensics' ?  This feels more like an 'add' than a 'replace'.

<!-- There are many many many use cases that could use GitBOMs.  An incomplete list would include:

* Detecting potential vulnerabilities in executables/containers.
* Identifying Open Source License obligations
* Identifying commercial license obligations
* More reliable attestation
* Post exploit forensics -->
{{< /home/section >}}

{{< home/section title="How?">}}

> GitBOM observes:

Reviewer: ```suggestion
Drawing on the version control system `git`, GitBOM observes that:
```

> 1. Every artifact is a [blob](/resources/glossay/#merkle-tree)

Reviewer: ```suggestion
1. Every artifact has a binary representation, a.k.a. a [blob](/resources/glossay/#merkle-tree)
```

2. Every blob has a [gitref](/resources/glossay/#merkle-tree)
3. Therefore the gitref may be used as an artifact id for an artifact

> 4. Most source code artifacts are stored already with their gitref as their id

Reviewer: ```suggestion
4. Most source code artifacts are stored with their gitref as their on-disk identifier
```

Author: I don't quite follow the 'on-disk' part of this.  I can totally get behind expanding 'id' to 'identifier' ... but curious what the thought process was with 'on-disk'

Reviewer: I was thinking of different ways that executable objects are identified on-disk/at-rest versus when in use. On second look, I think my own suggestion adds more confusion here than clarity.
